npm_and_yarn/spec/dependabot/npm_and_yarn/package_manager_helper_spec.rb

```diff
@@ -118,4 +118,58 @@
       end
     end
   end
+
+  describe "#installed_version" do
+    before do
+      allow(Dependabot::NpmAndYarn::Helpers).to receive_messages(
+        npm_version_numeric: 7,
+        yarn_version_numeric: 1,
+        pnpm_version_numeric: 7
+      )
+    end
+
+    context "when the installed version matches the expected format" do
+      before do
+        allow(Dependabot::SharedHelpers).to receive(:run_shell_command)
+          .with("npm --version", fingerprint: "<name> --version").and_return("7.5.2")
+      end
+
+      it "returns the raw installed version and caches it" do
+        expect(helper.installed_version("npm")).to eq("7.5.2")
+        # Check that the version is memoized
+        expect(helper.instance_variable_get(:@installed_versions)["npm"]).to eq("7.5.2")
```

Comment:
🤔 Unless we've identified this as a performance bottleneck, I'm not sure of the value we get from checking this. The benefit I do see, which might not need to be repeated, is that we have a guarantee that `@installed_versions` hasn't been typoed.